File: csharp/secret-handshake/SecretHandshake.cs
public static class SecretHandshake
{
    public static string[] Commands(int commandValue)
    {
        var actions = new List<string>();
        if ((commandValue & 1) == 1)
            actions.Add("wink");
        if ((commandValue & 2) == 2)
            actions.Add("double blink");
        if ((commandValue & 4) == 4)
            actions.Add("close your eyes");
        if ((commandValue & 8) == 8)
            actions.Add("jump");
        if ((commandValue & 16) == 16)
            actions.Reverse();
        return actions.ToArray();
    }
}

File: typescript/secret-handshake/secret-handshake.ts
type HandshakeAction = 'wink' | 'double blink' | 'close your eyes' | 'jump';

export function commands(code: number): HandshakeAction[] {
  const HANDSHAKE_ACTIONS: HandshakeAction[] = [
    'wink', 'double blink', 'close your eyes', 'jump'
  ];
  const actions: HandshakeAction[] = [];

  HANDSHAKE_ACTIONS.forEach((action, index) => {
    if ((code & (1 << index)) !== 0) {
      actions.push(action);
    }
  });
  if ((code & (1 << 4)) !== 0) {
    actions.reverse();
  }
  return actions;
}
